### PR TITLE
Add `report_body` member to `Quote3`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
The report body is now parsed and stored in the `Quote3` as a member.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #193** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/193" target="_blank">Graphite</a> 
    * **PR #194** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/194" target="_blank">Graphite</a> 
      * **PR #195** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/195" target="_blank">Graphite</a> 
    * **PR #196** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/196" target="_blank">Graphite</a> 
      * **PR #197** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/197" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/197?utm_source=stack-comment).

File: dcap/types/src/quote3.rs [hunk 1]
```diff
@@ -18,6 +18,9 @@ const QUOTE_SIZE: usize =
 // The offset to the authentication data
 const AUTH_DATA_OFFSET: usize = QUOTE_SIZE;
 
+// The offset to the report body for the app. From the start of the quote.
+const REPORT_BODY_OFFSET: usize = mem::size_of::<sgx_quote_header_t>();
```
Comment (jcape) [inline dcap/types/src/quote3.rs:22]: Is `sgx_quote_header_t` packed (or at least the same size on all the platforms we care to support)?

File: dcap/types/src/quote3.rs [hunk 2]
```diff
@@ -66,9 +69,10 @@ impl Error {
 type Result<T> = ::core::result::Result<T, Error>;
 
 /// Quote version 3
-#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
```
Comment (jcape) [inline dcap/types/src/quote3.rs:72]: I'd suggest we
1. Derive `Ord, PartialOrd` on `ReportBody`, and 
2. Implement Eq, PartialEq, Ord, PartialOrd manually on this type, using only the bytes (no sense comparing the same data twice)

**Review by jcape — APPROVED** [review]
Couple minor suggestions, otherwise LGTM

**Review by samdealy — APPROVED** [review]

File: dcap/types/src/quote3.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply dcap/types/src/quote3.rs:22]: Yes the `sgx_quote_header_t` is packed, it's inside of the `#pragma pack(push, 1)` in the header https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/quote_wrapper/common/inc/sgx_quote_3.h#L68

**Review by samdealy — APPROVED** [review]

**Review by jcape — APPROVED** [review]
LGTM, ty

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/197?utm_source=stack-comment).


Job Reference: yIELMGRp3yYkMM34LUYh

**Pull request merged**